### PR TITLE
chore: use uv for project management

### PR DESCRIPTION
**Pull request by bgrnwd**
### Description

`uv` 😎 

### Checklist

- [x] Code compiles correctly
- [x] Created tests which fail without the change (if applicable)
- [x] All lint and unit tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: .github/workflows/pypi-publish.yml [hunk 1]
```diff
@@ -2,7 +2,6 @@ name: Upload Python Package
 
 on:
   workflow_dispatch:
-    branches: [main]
```
Comment (tyler-richardett) [inline .github/workflows/pypi-publish.yml:5]: Ooh, do you not need to specify a branch for this any more? (And if so, should you remove the `:` after `workflow_dispatch`?)

**Review by tyler-richardett — APPROVED** [review]
lgtm

File: .github/workflows/pypi-publish.yml (hunk 1 above)
Comment (bgrnwd) [reply .github/workflows/pypi-publish.yml:5]: I don't think it was ever valid syntax, my IDE yelled at me. I think the `:` is still needed though?

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onworkflow_dispatch

Comment (tyler-richardett) [reply .github/workflows/pypi-publish.yml:5]: 🤷‍♂️

**Pull request merged**